cli/injector/main.go

```diff
@@ -630,16 +633,49 @@ func watchTargetAndReinject(deadline time.Time, commandName string, pulseActiveD
 			if !ok {
 				channel = nil
 
-				break
+				continue
```

Comment:
This is mostly a no-op, but I think it tracks what we want to do more accurately. It isn't that we want to leave the `select`, but that we want to return to the top of the `for`, to re-create the channel